examples/graph/tool_requires/using_protobuf.rst

```diff
@@ -0,0 +1,230 @@
+
+    protobuf_generate_cpp(PROTO_SRCS PROTO_HDRS addressbook.proto)
+
+    add_library(myserver src/myserver.cpp ${PROTO_SRCS})
+    target_include_directories(myserver PUBLIC include)
+
+    target_include_directories(myserver PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+      $<INSTALL_INTERFACE:include>
+    )
+
+    target_link_libraries(myserver PUBLIC protobuf::libprotobuf)
+
+    set_target_properties(myserver PROPERTIES PUBLIC_HEADER "include/myserver.h;${PROTO_HDRS}")
+    install(TARGETS myserver)
+
+To test that everything is working as expected, it includes a *test_package* folder where the *example.cpp* includes the
+generated *addressbook.pb.h* header:
+
+.. code-block:: cpp
+
+    #include <iostream>
+    #include <fstream>
+    #include <string>
+    #include "addressbook.pb.h"
+
+
+    int main(int argc, char* argv[]) {
+
+      GOOGLE_PROTOBUF_VERIFY_VERSION;
+
+
+      tutorial::AddressBook address_book;
+      auto * person = address_book.add_people();
+      person->set_id(1337);
+      std::cout << "Test(): created a person with id 1337\n";
+      // Optional:  Delete all global objects allocated by libprotobuf.
+      google::protobuf::ShutdownProtobufLibrary();
+
+      return 0;
+    }
+
+
+So, let's see if it works fine:
+
+.. code-block:: shell
+
+    $ conan create . --build missing
+    ...
+
+    Requirements
+        myserver/1.0#71305099cc4dc0b08bb532d4f9196ac1:c4e35584cc696eb5dd8370a2a6d920fb2a156438 - Build
+        protobuf/3.18.1#ac69396cd9fbb796b5b1fc16473ca354:e60fa1e7fc3000cc7be2a50a507800815e3f45e0#0af7d905b0df3225a3a56243841e041b - Cache
+        zlib/1.2.13#13c96f538b52e1600c40b88994de240f:d0599452a426a161e02a297c6e0c5070f99b4909#69b9ece1cce8bc302b69159b4d437acd - Cache
+    Build requirements
+        protobuf/3.18.1#ac69396cd9fbb796b5b1fc16473ca354:e60fa1e7fc3000cc7be2a50a507800815e3f45e0#0af7d905b0df3225a3a56243841e041b - Cache
+    ...
+
+    -- Install configuration: "Release"
+    -- Installing: /Users/myuser/.conan2/p/b/myser03f790a5a5533/p/lib/libmyserver.a
+    -- Installing: /Users/myuser/.conan2/p/b/myser03f790a5a5533/p/include/myserver.h
+    -- Installing: /Users/myuser/.conan2/p/b/myser03f790a5a5533/p/include/addressbook.pb.h
+
+    myserver/1.0: package(): Packaged 2 '.h' files: myserver.h, addressbook.pb.h
+    myserver/1.0: package(): Packaged 1 '.a' file: libmyserver.a
+    ....
+
+    ======== Testing the package: Executing test ========
+    myserver/1.0 (test package): Running test()
+    myserver/1.0 (test package): RUN: ./example
+    Test(): created a person with id 1337
+
+
+After seeing it's running OK, let's try to use cross-building. Notice that this part is based on MacOS Intel systems,
+and cross-compiling for MacOS ARM ones, but you could use your own profiles depending on your needs for sure.
+
+.. code-block:: shell
+
+    $ conan create . --build missing -pr:b host-x86_64 -pr:h host-armv8
```

Comment:
Maybe this is very OSX specific, a clearer warning that those profiles are Mac only might be good